compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java

```diff
@@ -469,12 +471,24 @@ private boolean withinRange(LineRange range, LineRange specifiedRange) {
     }
 
     private void addToCompiledSymbols(Set<Symbol> compiledSymbols, Scope.ScopeEntry scopeEntry, Location cursorPos,
-                                      Name name, BSymbol symbolEnvScopeOwner, Set<DiagnosticState> states) {
+                                      Name name, SymbolEnv symbolEnv, Set<DiagnosticState> states) {
         if (scopeEntry == null || scopeEntry.symbol == null || isFilteredVarSymbol(scopeEntry.symbol, states)) {
             return;
         }
 
+        BSymbol symbolEnvScopeOwner = symbolEnv.scope.owner;
         BSymbol symbol = scopeEntry.symbol;
+
+        // Checks if the enclosed node is within a worker declaration body and the encountered symbol is the same
+        // worker that is being declared.
+        if (Symbols.isFlagOn(symbolEnvScopeOwner.flags, Flags.WORKER)
+                && Symbols.isFlagOn(symbol.flags, Flags.WORKER)
+                && symbolEnv.enclEnv != null
+                && symbolEnv.enclEnv.node.getKind() == NodeKind.FUNCTION
+                && isWithinCurrentWorker(((BLangFunction) symbolEnv.enclEnv.node).defaultWorkerName, symbol)) {
+            return;
+        }
```

Comment:
What if we enclose this whole if check within `isWithinCurrentWorker` so the logic will be much cleaner? WDYT? Other than that, the changes looks good

Reply:
@nadeeshaan updated.